src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -600,6 +602,80 @@ class ChooseStringGenerator : public Generator<std::string> {
     bool _deterministic;
 };
 
+/**
+ * FrequencyMaps
+ * A list of {string, count} pairs.
+ * An item will be pulled from a random bucket and its count will be decremented. If all buckets are
+ * empty, the generator throws an error.
+ */
+class FrequencyMapGenerator : public Generator<std::string> {
+public:
+    FrequencyMapGenerator(const Node& node, GeneratorArgs generatorArgs) : _rng{generatorArgs.rng} {
+        if (!node["from"].isMap()) {
+            std::stringstream msg;
+            msg << "Malformed node for 'TakeRandomStringFromFrequencyMap' from a map " << node;
+            BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax(msg.str()));
+        }
+
+        for (const auto&& [k, v] : node["from"]) {
+            _map.push_back(v.key(), static_cast<size_t>(v.maybe<std::int64_t>().value()));
```

Comment:
1. I'm curious if `.maybe<size_t>()` works? If not no biggie but maybe we should make it work with a converter (can be a separate PR and this is fine for now either way).
2. If you're going to `.value()` the `optional` may as well just use `.to`: `v.to<int64>()` since you don't want/need to care about the case where it's unspecified.